ntbs-service-unit-tests/DataAccess/AlertRepositoryTests.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using ntbs_service.DataAccess;
+using ntbs_service.Models.Entities;
+using ntbs_service.Models.Entities.Alerts;
+using ntbs_service.Models.Enums;
+using ntbs_service.Models.ReferenceEntities;
+using Xunit;
+
+namespace ntbs_service_unit_tests.DataAccess
+{
+    public class AlertRepositoryTests : IClassFixture<RepositoryFixture<AlertRepository>>, IDisposable
+    {
+        private readonly NtbsContext _context;
+        private readonly AlertRepository _alertRepo;
+
+        public AlertRepositoryTests(RepositoryFixture<AlertRepository> alertRepositoryFixture)
+        {
+            _context = alertRepositoryFixture.Context;
+            _alertRepo = new AlertRepository(_context);
+        }
+
+        private readonly User _caseManager = new User {DisplayName = "Pharoah Sanders", Username = "ps@fake-nhs.co"};
+        private readonly TBService _tbService = new TBService {Code = "TB-TEST-001", Name = "Test Service 001"};
+
+        [Theory]
+        [InlineData(true, true)]
+        [InlineData(false, true)]
+        [InlineData(true, false)]
+        [InlineData(false, false)]
+        public async void CorrectlyMapsTransferAlertToDisplayAlert(bool hasCaseManager, bool hasTbService)
+        {
+            // Arrange
+            var notification = await GivenNotificationWithCaseManagerAndTbService
+                (hasCaseManager ? _caseManager : null, hasTbService ? _tbService : null);
+            await _context.Alert.AddAsync(new TransferAlert()
+            {
+                CaseManagerId = notification.HospitalDetails.CaseManagerId,
+                TbServiceCode = notification.HospitalDetails.TBServiceCode,
+                CreationDate = new DateTime(2004,2,2),
+                NotificationId = notification.NotificationId,
+                AlertType = AlertType.TransferRequest
+            });
+            _context.SaveChanges();
+
+            // Act
+            var result = (await _alertRepo.GetOpenAlertsForNotificationAsync(notification.NotificationId)).Single();
+
+            // Assert
+            Assert.Equal(hasCaseManager ? _caseManager.DisplayName : null, result.CaseManagerName);
```

Comment:
In general, it's best to avoid logic inside your assertions. It makes your tests harder to maintain, and it also makes it possible to replicate a bug from your code inside your tests (so that the tests still pass).

Instead, it's better to have extra parameters for expected values in the test definition.

Reply:
This goes for the other tests in this class too.